### PR TITLE
Transformations: Handle undefined values when calculating mean and total in add field from calculations

### LINKED ISSUE #78644: Transformations: add field from calculation breaks when calculating cumulative total on data sets containing undefined
### What happened?

When calculating cumulative total on datasets with undefined values the transformation tries to add previous total with undefined resulting in `NaN`

### What did you expect to happen?

undefined values to be counted as 0

### Did this work before?

no

### How do we reproduce it?

1. have a data set with undefined values
2. add field from calculation transformation
3. switch mode to cumulative total

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

_No response_

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by oscarkilhed**
**What is this feature?**

Handles undefined values the same as nulls to avoid adding undefined to the calculated total and ending up with `NaN`

**Which issue(s) does this PR fix?**:

Fixes #78644

### CODE REVIEW

**Review by mdvictor — APPROVED** [review]

File: packages/grafana-data/src/transformations/transformers/calculateField.ts [hunk 1]
```diff
@@ -315,22 +315,22 @@ function getCenteredWindowValues(frame: DataFrame, reducer: ReducerID, selectedF
...
-          if (selectedField.values[last] !== null) {
+          if (selectedField.values[last] !== null && selectedField.values[last] !== undefined) {
             sum += selectedField.values[last];
             count++;
           }
         }
         if (first > 0) {
           // Remove values that have fallen outside of the window, if the start of the window isn't outside of the data.
-          if (selectedField.values[first - 1] !== null) {
+          if (selectedField.values[first - 1] !== null && selectedField.values[first] !== undefined) {
```
Comment (leeoniya) [inline packages/grafana-data/src/transformations/transformers/calculateField.ts:333]: first - 1?

**Comment by leeoniya** [comment]
can we just use `!= null` here.these duplicate checks are slower, and harder to read :)

**Comment by oscarkilhed** [comment]
> can we just use `!= null` here. these duplicate checks are slower, and harder to read :)

Will do!

File: packages/grafana-data/src/transformations/transformers/calculateField.ts (hunk 1 above)
Comment (oscarkilhed) [reply packages/grafana-data/src/transformations/transformers/calculateField.ts]: Oh my, that was sloppy of me.

Should have added some tests for the undefined case from the start... Anyway, they're added now.

**Pull request merged**